_posts/2022-09-20-internship-projects-2022-concrete-playback.md

```diff
@@ -0,0 +1,239 @@
+---
+layout: post
+title:  "Internship Projects 2022: Concrete Playback"
+---
+
+Today we're starting a series of posts about the internship projects carried out
+in our team during 2022. The Kani team is proud to be part of the AWS Automated
+Reasoning teams, which every year hosts a number of interns to work on
+automated-reasoning projects for tools like Kani. More details on AWS Automated
+Reasoning areas of work and available locations can be found here. If you're a
+Masters or PhD interested in Automated Reasoning, please consider applying to
+the following openings:
+ * [2023 Applied Science Internship (Master's student)](https://www.amazon.jobs/en/internal/jobs/2173429/2023-applied-science-internship-automated-reasoning-united-states)
+ * [2023 Applied Science Internship (PhD student)](https://amazon.jobs/en/internal/jobs/1882106/applied-scientist-automated-reasoning-group)
+
+> This internship project was executed by 
+[Sanjit Bhat](https://sanjit-bhat.github.io/). Sanjit joined the Kani team as an SDE Intern
+while finishing his undergraduate studies at [UT
+Austin](https://www.utexas.edu/) and is now a PhD student at
+[MIT](https://www.mit.edu/). We wish him the best in his PhD studies!
+
+## Proof Debugging
+
+We commonly use the term **Proof Debugging** to refer to the process of
+debugging failing proofs that we expected to succeed.
+The process isn't very different to debugging unit tests: There's one or more errors
+due to assumptions made about the function under test, and how the inputs and/or outputs
+relate to that function. The goal then is to find what assumption was made and
+correct it accordingly.
+
+Let's talk about how tooling can make this process easier for users.
+
+### CBMC Viewer
+
+As you may know, Kani uses [CBMC](https://github.com/diffblue/cbmc) to perform
+the main analysis required for verification. When one or more checks have failed,
+that's a verification failure. In these cases, CBMC can return a text-based
+trace which includes the sequence of steps leading to the failed check. We often
+call this a (proof) counter-example.
```

Comment:
I think it may be better to just use the phrase counter-example, since we are explicitly talking about viewing traces and debugging traces. Proofs in my mind are positive and not negative counter examples. Would simplify it for the reader... suggestion. not required.

Reply:
Makes sense. Removed "(proof)" from there. I'm still using the term "trace" throughout the post.